### ISSUE TITLE
False positive: walrus operator probably triggers chained-comparison warning

### ISSUE THREAD

**Issue by algonell** [PyCQA/pylint#7811]
### Bug description

Comparison followed by walrus operator and comparison (both use a shared variable - j) suggests to chain the comparisons although they are not related. 

```python
j = i

while j >= 0 and (val := self.df['3/10'].values[j]) < 0:
    hist.appendleft(val)
    j -= 1
```

### Command used

```shell
pylint futures_utils.py
```

### Pylint output

```shell
************* Module futures_utils
Algonell/futures_utils.py:1750:18: R1716: Simplify chained comparison between the operands (chained-comparison)
```

### Expected behavior

No warning on this line.

### Pylint version

```shell
pylint 2.15.6
astroid 2.12.12
Python 3.10.7 (main, Nov  2 2022, 18:49:29) [GCC 12.2.0]
```

### OS / Environment

Distributor ID:	Ubuntu
Description:	Ubuntu 22.10
Release:	22.10
Codename:	kinetic

**Comment by Pierre-Sassoulas** [comment]
Hello, thank you for opening the issue. Could you provide the code as text instead of as an image ? Text is crawl-able by search engines, readable by visually impaired persons and copy pastable to test the issue fast, while images are not.

**Comment by clavedeluna** [comment]
@algonell I think pylint is telling you to do this:

```
while (val := self.df['3/10'].values[j]) < 0 <= j:
    hist.appendleft(val)
    j -= 1
```

In which case the warning is correct.

**Comment by algonell** [comment]
@clavedeluna you are absolutely right! I could not understand why the warning was only for one of the two "similar" cases:

```python
if val < 0:  # negative island
    # left
    j = i

    # pylint: disable-next=chained-comparison
    while j >= 0 and (val := self.df['3/10'].values[j]) < 0:
        hist.appendleft(val)
        j -= 1
else:  # positive island
    # left
    j = i

    while j >= 0 and (val := self.df['3/10'].values[j]) > 0:
        hist.appendleft(val)
        j -= 1
```

The first one is a proper chain of X < 0 <= Y while the second one is X > 0 <= Y which is not, thank you!

**Comment by clavedeluna** [comment]
Awesome! Glad to help! For the record, I took the code and simplified it a bunch, replaced it with lists, etc, just so I could understand it. Then I re-built it.

I think this means this issue can be closed. If you disagree, feel free to reopen.

**Comment by GastonLer** [comment]
Hello,
I would like chained-comparison to be disabled in this case where the two parts do not have the same "unit".
In this example J>=0 represents an index and val <0 represents a value with an unknown unit. For me, it's confusing to mix this two items.
Let's say, we want to raise an exception when a weak north-north-east wind occurs. We should write
```
if speed < 10 and 10<direction<40:
 raise Exception("weak NNE wind")
```

speed is m.s-1 unit
direction is degree unit.
so if we write
```
if speed <10 <direction<40:
 raise Exception("weak NNE wind")
```
it seems that we compare m.s-1 to degree : physically chocking
if the item in the middle is a variable : it's developper responsablity to check the coherence.
but if the item in the middle is a constant, the rule shouldn't have to be applied.

What's your opinion ?

**Comment by Pierre-Sassoulas** [comment]
You do have a point but there's no way to infer the physical unit of a variable (I tried to use the semantical analysis libs in the past, but 1) it's not convincing and 2) they are not very popular so i'm not sure we should check the typing all the time for such a low rate of having a better check if they are actually used).

Imo the only solution here is for the user to add a local disable (or a comment clarifying the unit mismatch).
